presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergSplitManager.java

```diff
@@ -57,7 +57,12 @@ public ConnectorSplitSource getSplits(
         HiveMetastore metastore = transactionManager.get(transaction).getMetastore();
         Table icebergTable = getIcebergTable(metastore, hdfsEnvironment, session, table.getSchemaTableName());
 
-        TableScan tableScan = getTableScan(table.getPredicate(), table.getSnapshotId(), icebergTable);
+        TableScan tableScan = getTableScan(
+                // TODO: Remove TupleDomain#simplify once Iceberg supports IN expression
+                table.getEnforcedPredicate()
+                        .intersect(table.getPredicate()).simplify(),
```

Comment:
Should this move inside `toIcebergExpression()`?

Reply:
`simplify` needs to be used cautiously only when required because it can change the predicate.

For example, it cannot be used for an "enforced" constraint. we may end up returning incorrect results for filters on partition columns for SELECT queries, or deleting undesired partitions in case of metadata delete.